comment/mixins.py

```diff
@@ -1,113 +1,103 @@
-from django.contrib.auth.mixins import LoginRequiredMixin
-from django.contrib.contenttypes.models import ContentType
+import abc
+
+from django.contrib.auth.mixins import LoginRequiredMixin, AccessMixin
+from django.core.exceptions import ImproperlyConfigured
+from django.http import HttpResponseBadRequest, HttpResponseForbidden
 from django.utils.translation import gettext_lazy as _
 
-from comment.models import Comment
-from comment.utils import get_data_for_request, get_response_for_bad_request
+from comment.conf import settings
+from comment.utils import is_comment_admin, is_comment_moderator
+from validators import ValidatorMixin
+
+
+class AJAXRequiredMixin:
+    def dispatch(self, request, *args, **kwargs):
+        if not request.META.get('HTTP_X_REQUESTED_WITH', None) == 'XMLHttpRequest':
+            return HttpResponseBadRequest(_('Only AJAX request are allowed'))
+        return super().dispatch(request, *args, **kwargs)
+
 
+class BasePermission(AJAXRequiredMixin):
+    def has_permission(self, request):
+        return True
 
-class CommentUpdateMixin(LoginRequiredMixin):
+    def has_object_permission(self, request, obj):
+        return True
```

Comment:
Inheriting `AJAXRequiredMixin` here means that only AJAX requests will now be allowed even for DRF?

Reply:
These mixins are not used in the API, we are using the default permission. I prefer to keep this logic separate to avoid complexity. However, some api views need to be updated to have the same functionality implemented in the GUI version i.e. deleting comment by admin.